staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go

```diff
@@ -157,3 +157,113 @@ type TracingConfiguration struct {
 	// Embed the component config tracing configuration struct
 	tracingapi.TracingConfiguration
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AuthenticationConfiguration provides versioned configuration for authentication.
+type AuthenticationConfiguration struct {
+	metav1.TypeMeta
+
+	// jwt is a list of authenticator to authenticate Kubernetes users using
```

Comment:
It is probably easier to drop the docs on the internal types since I think we will go through a few iterations of the verbiage anyway.